functions/zoom-uploader.py

```diff
@@ -310,3 +394,28 @@ def _generate_presigned_url(self, s3_filename):
         )
         logger.info("Got presigned url {}".format(url))
         return url
+
+    def generate(self):
+        for primary_view, secondary_views in self.VIEW_PRIORITES.items():
+            if not self._has_view(primary_view) or self._has_presenter():
+                continue
+            self._add_presenter(primary_view)
+
+            for view in secondary_views:
+                if not self._has_view(view):
+                    continue
+                self._add_secondary(view)
+
+        if self.num_files == 0:
```

Comment:
Shouldn't this just be `if not self._has_presenter` ? At this point. If it already has a presenter then all of the fallback options would have already been checked for secondaries.